ProcessLib/SmallDeformation/SmallDeformationProcess.h

```diff
@@ -0,0 +1,176 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESS_LIB_SMALLDEFORMATIONPROCESS_H_
+#define PROCESS_LIB_SMALLDEFORMATIONPROCESS_H_
+
+#include <cassert>
+
+#include "NumLib/Extrapolation/LocalLinearLeastSquaresExtrapolator.h"
+#include "ProcessLib/Process.h"
+#include "ProcessLib/SmallDeformation/CreateLocalAssemblers.h"
+
+#include "SmallDeformationFEM.h"
+#include "SmallDeformationProcessData.h"
+
+namespace ProcessLib
+{
+namespace SmallDeformation
+{
+template <int DisplacementDim>
+class SmallDeformationProcess final : public Process
+{
+    using Base = Process;
+
+public:
+    SmallDeformationProcess(
+        MeshLib::Mesh& mesh,
+        Base::NonlinearSolver& nonlinear_solver,
+        std::unique_ptr<Base::TimeDiscretization>&& time_discretization,
+        std::unique_ptr<NumLib::ConvergenceCriterion>&& convergence_criterion,
+        std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+        std::vector<std::reference_wrapper<ProcessVariable>>&&
+            process_variables,
+        SmallDeformationProcessData<DisplacementDim>&& process_data,
+        SecondaryVariableCollection&& secondary_variables,
+        ProcessOutput&& process_output,
+        NumLib::NamedFunctionCaller&& named_function_caller)
+        : Process(mesh, nonlinear_solver, std::move(time_discretization),
+                  std::move(convergence_criterion), parameters,
+                  std::move(process_variables), std::move(secondary_variables),
+                  std::move(process_output), std::move(named_function_caller)),
+          _process_data(std::move(process_data))
+    {
+        if (dynamic_cast<NumLib::ForwardEuler*>(
+                &Base::getTimeDiscretization()) != nullptr)
+        {
+            ERR("The SmallDeformationProcess can not be solved with the "
+                "ForwardEuler time discretization scheme. Aborting");
+            // Because the M matrix is not assembled. Thus, the linearized
+            // system would be singular. The same applies to CrankNicolson with
+            // theta = 0.0, but this case is not checked here.  Anyway, the
+            // SmallDeformationProcess shall be transferred to a simpler
+            // ODESystemTag in the future.
+            std::abort();
```

Comment:
For dynamic deformation, M is from the second derivative of displacement with respect to time. Therefore if-condition of (dynamic_cast<NumLib::ForwardEuler*> is not adequate for this case.